platform/osx/app/MapDocument.m

```diff
@@ -457,8 +488,8 @@ - (BOOL)validateMenuItem:(NSMenuItem *)menuItem {
     if (menuItem.action == @selector(dropManyPins:)) {
         return YES;
     }
-    if (menuItem.action == @selector(removeAllPins:)) {
-        return self.mapView.annotations.count;
+    if (menuItem.action == @selector(removeAllAnnotations:)) {
+        return self.mapView.annotations.count ? YES : NO;
```

Comment:
Did this because `self.mapView.annotations.count` alone wasn't doing the job and `YES`/`NO` is pleasing semantically.